Registry/RegistryParse.py

```diff
@@ -588,12 +588,14 @@ def decode_utf16le(s):
       catch any Unicode-related exceptions, such as UnicodeDecodeError,
       so these should be handled by the caller.
 
-    @type s: str
+    @type s: bytes
     @param s: a bytestring to pase
     @rtype: unicode
     @return: the unicode string decoded from `s`
     @raises: this function does not attempt to catch any Unicode-related exception, so the caller should handle these.
     """
+    if s is None:
+        return None
```

Comment:
Please do not return `None` from a function that otherwise returns a specific type (`str` in this case). Raise exceptions to signal errors.

The reason is that by returning `None`, we'd have to add checks `if ret is None:` every place these functions are called, in addition to the existing `try`/`except` handlers. Its also easy to forget these checks, and then you get `AttributeErrors`, etc. on mysterious `None` values if there's ever a bug.